### PR TITLE
added get ips for filtering

### PR DESCRIPTION
**Pull request by benni043**
added get ips

### CODE REVIEW

File: src/entry/logEntry.store.ts [hunk 1]
```diff
@@ -20,59 +23,30 @@ export class LogEntryStore {
...
-          queryString += "AND content = $" + queryParams.length;
-        }
-      }
-
-      if (logEntryRequestData.filters?.classification) {
-        queryParams.push(logEntryRequestData.filters.classification);
-        queryString += "AND classification = $" + queryParams.length;
-      }
+      if (logEntryRequestData.filters) {
+        const queryData = this.applyFilters(
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
        const filteredQueryData = this.applyFilters(
```

File: src/entry/logEntry.store.ts [hunk 2]
```diff
@@ -20,59 +23,30 @@ export class LogEntryStore {
...
-        } else {
-          queryParams.push(logEntryRequestData.filters.date.from);
-          queryParams.push(logEntryRequestData.filters.date.to);
-
-          queryString += `AND creation_date BETWEEN $${
-            queryParams.length - 1
-          } AND $${queryParams.length}`;
-        }
+        queryParams = queryData.queryParams;
+        queryString = queryData.queryString;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
        queryParams = filteredQueryData.queryParams;
        queryString = filteredQueryData.queryString;
```

File: src/entry/logEntry.store.ts [hunk 3]
```diff
@@ -95,7 +69,95 @@ export class LogEntryStore {
...
+      let queryParams: any[] = [sessionID, ipRequestData.files];
+      let queryString = `
+      SELECT service_ip
+            FROM loggaroo.log_entry
+              WHERE session_id = $1
+                AND file_name = ANY($2)
+                `;
+
+      if (ipRequestData.filters) {
+        const queryData = this.applyFilters(
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
        const filteredQueryData = this.applyFilters(
```

File: src/entry/logEntry.store.ts [hunk 4]
```diff
@@ -95,7 +69,95 @@ export class LogEntryStore {
...
+          ipRequestData.filters,
+          queryString,
+          queryParams,
+        );
+
+        queryParams = queryData.queryParams;
+        queryString = queryData.queryString;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
        queryParams = filteredQueryData.queryParams;
        queryString = filteredQueryData.queryString;
```

File: src/entry/logEntry.store.ts [hunk 5]
```diff
@@ -95,7 +69,95 @@ export class LogEntryStore {
...
+        .query<IP>(query);
+
+      const ipObjs = result.rows;
+
+      const ips: string[] = [];
+      for (const ip of ipObjs) {
+        ips.push(ip.ip);
+      }
+
+      return ips;
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts]:
```suggestion
      return result.rows.map((e) => e.ip);
```

File: src/entry/logEntry.store.ts [hunk 6]
```diff
@@ -95,7 +69,95 @@ export class LogEntryStore {
...
+    } catch (e) {
+      console.log(e);
+      return RequestError.serverError;
     }
   }
+
+  applyFilters(filters: Filters, queryString: string, queryParams: any[]) {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.store.ts:116]: This code modifies the original queryParams array I think. So clone the array before using it:
```suggestion
  applyFilters(filters: Filters, queryString: string, queryParams: any[]) {
      queryParams = [...queryParams];
```

File: src/entry/logEntry.ts [hunk 7]
```diff
@@ -13,6 +13,10 @@ export type LogEntry = {
   content: string;
 };
 
+export type IP = {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.ts]: To be correct it should be:
```suggestion
export type Ip = {
```

File: src/entry/logEntry.ts [hunk 8]
```diff
@@ -28,6 +32,11 @@ export type LogEntryRequestData = {
   filters: Filters | undefined;
 };
 
+export type IPRequestData = {
```
Comment (JonasPfeifer05) [inline src/entry/logEntry.ts]: To be correct it should be:
```suggestion
export type IpRequestData = {
```

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]

File: src/entry/logEntry.service.ts [hunk 9]
```diff
@@ -20,24 +21,31 @@ export class LogEntryService {
...
       }
 
       return await LogEntryStore.getInstance().get(sessionID, logEntryData);
     } catch (e) {
-      return LogEntryRequestError.wrongBodyData;
+      return RequestError.wrongBodyData;
     }
   }
+
+  async getIPs(
```
Comment (xFizzar) [inline src/entry/logEntry.service.ts]:
```suggestion
  async getIps(
```

**Review by xFizzar — DISMISSED** [review]

**Review by JonasPfeifer05 — APPROVED** [review]
Ok

**Pull request merged**